vernac/metasyntax.ml

```diff
@@ -1222,12 +1213,32 @@ let check_locality_compatibility local custom i_typs =
                     strbrk " which is local."))
       (List.uniquize allcustoms)
 
-let compute_syntax_data local df modifiers =
+let warn_deprecated_compat =
+  CWarnings.create ~name:"deprecated-compat" ~category:"deprecated"
+    (fun () -> Pp.(str"The \"compat\" modifier is deprecated." ++ spc () ++
+                   str"Please use the \"deprecated\" attributed instead."))
+
+(* Returns the new deprecation and the onlyparsing status. This should be
+removed together with the compat syntax modifier. *)
+let merge_compat_deprecation compat deprecation =
+  match compat, deprecation with
+  | Some Flags.Current, _ -> deprecation, true
+  | Some _, Some _ ->
+    CErrors.user_err Pp.(str"The \"compat\" modifier cannot be used with the \"deprecated\" attribute."
+                        ++ spc () ++ str"Please use only the latter.")
+  | Some v, None ->
+    warn_deprecated_compat ();
+    Some (Deprecation.make ~since:(Flags.pr_version v) ()), true
```

Comment:
Previously, the warning was:

```
Notation foo := plus (compat "8.8").
Check foo. (* foo is Nat.add [compatibility-notation,deprecated] *)
```

Now it will be `Notation foo is deprecated since 8.8.`

Keeping the warning message unchanged is going to be unnecessarily tedious but at least, it should include a note giving the qualid to replace the notation.

Reply:
This is going to require significant changes in the API, just for the compat layer. Who's using `compat`?